pkg/login/social/social.go

```diff
@@ -389,29 +410,18 @@ func (s *SocialBase) extractRoleAndAdmin(rawJSON []byte, groups []string, legacy
 		}
 	}
 
-	return s.defaultRole(legacy), false
+	return "", false
 }
 
 // defaultRole returns the default role for the user based on the autoAssignOrgRole setting
 // if legacy is enabled "" is returned indicating the previous role assignment is used.
-func (s *SocialBase) defaultRole(legacy bool) org.RoleType {
-	if s.roleAttributeStrict {
-		s.log.Debug("RoleAttributeStrict is set, returning no role.")
-		return ""
-	}
-
-	if s.autoAssignOrgRole != "" && !legacy {
+func (s *SocialBase) defaultRole() org.RoleType {
+	if s.autoAssignOrgRole != "" {
 		s.log.Debug("No role found, returning default.")
 		return org.RoleType(s.autoAssignOrgRole)
 	}
 
-	if legacy && !s.skipOrgRoleSync {
-		s.log.Warn("No valid role found. Skipping role sync. " +
-			"In Grafana 10, this will result in the user being assigned the default role and overriding manual assignment. " +
-			"If role sync is not desired, set skip_org_role_sync for your provider to true")
-	}
-
-	return ""
+	return org.RoleViewer
```

Comment:
Question: should we also set the default value to `Viewer` [here](https://github.com/grafana/grafana/blob/main/pkg/setting/setting.go#L1641)? `Viewer` is the default in config.ini. I have a vague memory of docs mentioning `Editor` as the default, but I can't find it now.